test/parser/test_errors.rb

```diff
@@ -1,58 +1,62 @@
 require File.expand_path("../../helpers", __FILE__)
 
 class ParserErrors < Test::Unit::TestCase
+  def setup
```

Comment:
Some of these tests depend on Parser ivars being set up beforehand.

The only reason they used to work without this setup was because these ivars lived on the class and their initialization leaked in from other tests calling ::parse. They never worked in isolation.

Reply:
Ouch. I totally missed that. It doesn't look like it broke any tests though, thankfully.

Reply:
I mean many, not any.